py/kubeflow/testing/cleanup_ci.py

```diff
@@ -317,38 +315,41 @@ def cleanup_instance_groups(args):
       for s in results["items"]:
         name = s["name"]
         age = getAge(s["creationTimestamp"])
+        size = s["size"]
+        if size > 0:
```

Comment:
Doesn't this make our script less restrictive?
Right now we GC IG's after they expire. You are now skipping IG's which have expired but have non-zero size. 

If IUC there were two sets of IG's that we were leaking before because the regex didn't match.
k8s-ig.*
gke-kf-vmaster.*

You've updated the regex to match these IGs  So that should take care of the leak.

So separate from that do you also want to guard against GC'ing IGs with size > 0 ?

Reply:
I think if the size > 0, we are not able to delete the IG anyway because of resource dependency.

To the scenario you described, cleanup_ci will first delete the cluster, making the size to 0, IG will be cleaned up during the same run or next.

Reply:
@gabrielwen Thanks. Looks like we do set the min pool size to zero
https://github.com/kubeflow/manifests/blob/00fa81b956682ee835d75e21413c8008bf2bf170/gcp/deployment_manager_configs/cluster-kubeflow.yaml#L53